src/ZipStream.php

```diff
@@ -251,297 +330,241 @@ public function __construct($name = null, $opt = array())
      * @throws \ZipStream\Exception\FileNotFoundException
      * @throws \ZipStream\Exception\FileNotReadableException
      */
-    public function addFileFromPath($name, $path, $opt = array(), $storage_method = "deflate")
+    public function addFileFromPath($name, $path, $opt = array(), $method = 'deflate')
     {
-        if (!is_readable($path)) {
-            if (!file_exists($path)) {
-                throw new FileNotFoundException($path);
-            }
-            throw new FileNotReadableException($path);
-        }
-
-        if ($this->isLargeFile($path)) {
-            // file is too large to be read into memory; add progressively
-            $this->addLargeFile($name, $path, $opt);
-        } else {
-            // file is small enough to read into memory; read file contents and
-            // handle with addFile()
-
-            $data = file_get_contents($path);
-            $this->addFile($name, $data, $opt, $storage_method);
-        }
+        $file = new File($this, $name, $opt, $method);
+        $file->processPath($path);
     }
 
     /**
-     * Is this file larger than large_file_size?
+     * addFile_from_stream
      *
-     * @param string $path
-     * @return Boolean
+     * dds an open stream to the archive uncompressed
+     *
+     * @param String $name - path of file in archive (including directory).
+     * @param Resource $stream - contents of file as a stream resource
+     * @param array $opt - Hash of options for file (optional, see "File Options" below).
+     *
+     * File Options:
+     *  time     - Last-modified timestamp (seconds since the epoch) of
+     *             this file.  Defaults to the current time.
+     *  comment  - Comment related to this file.
+     *
+     * Examples:
+     *
+     *   // create a temporary file stream and write text to it
+     *   $fp = tmpfile();
+     *   fwrite($fp, 'The quick brown fox jumped over the lazy dog.');
+     *
+     *   // add a file named 'streamfile.txt' from the content of the stream
+     *   $x->addFile_from_stream('streamfile.txt', $fp);
+     *
+     * @return void
      */
-    protected function isLargeFile($path)
+    public function addFileFromStream($name, $stream, $opt = array(), $method = 'deflate')
     {
-        $st = stat($path);
-        return ($this->opt[self::OPTION_LARGE_FILE_SIZE] > 0) && ($st['size'] > $this->opt[self::OPTION_LARGE_FILE_SIZE]);
+        $file = new File($this, $name, $opt, $method);
+        $file->processStream(new DeflateStream($stream));
     }
 
     /**
-     * Add a large file from the given path.
+     * addFile_from_Psr7Stream
+     *
+     * dds an open stream to the archive uncompressed
```

Comment:
`s/dds/adds/`

Reply:
Fixed. Also fixed underscores in method names.